### ISSUE TITLE
Longer pauses causes ingame time to advance, game freezes

### ISSUE THREAD

**Issue by copygirl** [anegostudios/VintageStory-Issues#330]
**Game Version:** 1.10.8
**Platform:** Arch Linux
**Modded:** CarryCapacity 0.4.4

### Description
Pausing the game for a longer amount of time freezes the game. If left paused for a few minutes, the game will freeze for a noticeable amount of time and, when done, in-game time will have advanced by a few in-game hours or so.

### Expected behavior
Pause the game.

### Logs
```
19.9.2019 20:29:04 [Notification] Server ticking has been suspended
19.9.2019 20:32:14 [Notification] A client reconnected, resuming game calendar.
19.9.2019 20:32:14 [Notification] Server ticking has been resumed
19.9.2019 20:32:18 [Warning] Server overloaded. A tick took 4485ms to complete.
19.9.2019 20:32:18 [Notification] Unloaded 224 chunks.
19.9.2019 20:32:19 [Notification] Server ticking has been suspended
19.9.2019 20:32:19 [Notification] Autosaving game world. Notifying mods, then systems of save...
19.9.2019 20:32:19 [Notification] Defragmented listener lists
19.9.2019 20:32:19 [Event] Mods and systems notified, now saving everything...
19.9.2019 20:32:20 [Event] Saved savegamedata...
19.9.2019 20:32:20 [Event] Saved map regions...
19.9.2019 20:32:20 [Event] Saved map chunks...
19.9.2019 20:32:21 [Event] Saved 300 chunks...
19.9.2019 20:32:21 [Event] Saved loaded chunks...
19.9.2019 20:32:21 [Event] World saved! Saved 344 chunks, 52 mapchunks, 2 mapregions.
19.9.2019 20:32:21 [Notification] A client reconnected, resuming game calendar.
19.9.2019 20:32:21 [Notification] Server ticking has been resumed
19.9.2019 20:32:21 [Warning] Server overloaded. A tick took 1555ms to complete.
19.9.2019 20:32:22 [Notification] Delayed save of 200 generating chunks done.
19.9.2019 20:32:29 [Notification] Unloaded 96 chunks.
```

**Comment by Angelnoir** [comment]
Issue also found by Gryphius218 but a bit different:
They paused the game for around 10 Mins, game freezed and after another 10 Minutes they died and were disconnected from the Game. They were still dead after loading the game.

**Comment by Bloodsaw** [comment]
Since I'm unable to reproducing this issue, maybe this happens only in situation of high resource contention? 
Depending on settings you need up to 6GB of RAM of free memory, so if your OS is starting to swapping memory to disk, especially if you tab out and start to multitask in modern browsers, this could lead to huge lag spikes if returning into the game cuz most of the RAM is now on disk.
I guess this will impact every PC with 8GB of RAM or lower. 

I will try to reproduce it on my notebook with only 4GB of RAM and swap file on an usb drive. 

`( Will add-edit the results here) `

**Comment by VallenFrostweaver** [comment]
I have 32GB of RAM and experience this as does my wife on her PC too. We have new PC's with relatively high stats and this lag spike for a long pause still happens.

**Comment by Bloodsaw** [comment]
I heard yesterday in discord that @tyronx already implemented a fix for this in 1.12.6.
As this is unconfirmed, I guess you should try to reproduce this issue with the newest update and report back here.
Since tyron is searching for people able to replicate this issue, he would be glad when you submit your client-main.txt and server-main.txt (from the host) after the issue occurred.

**Comment by tyronx** [comment]
Some more information on this. Its not yet fixed but I think I'm getting close to a fix. It seems the server (or client?) doesn't properly unpause, which causes various tasks to accumulate over time that are then all processed at once when the game gets unpaused.

Starting from v1.12.9, when the developer mode is active and extended debug info has been activated, the game will just straight up crash when any subsystem enqueues a task while the game is paused. This will get me a stack trace to find the culprit.

**Comment by tyronx** [comment]
I think we can safely close this now. This is fixed in v1.12.9 and I've heard of no new reports.
